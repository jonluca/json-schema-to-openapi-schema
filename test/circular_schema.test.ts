import convert from '../src';
import { getSchema } from './helpers';
const test = 'circular';

it(`converts ${test}/openapi.json`, async ({ expect }) => {
	const schema = getSchema(test + '/json-schema.json');
	const result = await convert(schema, {
		dereference: true,
		dereferenceOptions: { dereference: { circular: 'ignore' } },
	});
	const expected = getSchema(test + '/openapi.json');
	expect(result).toEqual(expected);
});

it(`converting ${test}/openapi.json in place`, async ({ expect }) => {
	const schema = getSchema(test + '/json-schema.json');
	const result = await convert(schema, {
		cloneSchema: false,
		dereference: true,
		dereferenceOptions: { dereference: { circular: 'ignore' } },
	});
	const expected = getSchema(test + '/openapi.json');
	expect(schema).toEqual(result);
	expect(result).toEqual(expected);
});

it(`converting ${test}/openapi.json without circular references turned off `, async ({
	expect,
}) => {
	const schema = getSchema(test + '/json-schema.json');
	const result = await convert(schema, {
		cloneSchema: false,
		dereference: true,
	});
	expect(result).toMatchSnapshot();
});
