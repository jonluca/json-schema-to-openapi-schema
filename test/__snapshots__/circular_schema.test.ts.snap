// Vitest Snapshot v1

exports[`converting circular/openapi.json without circular references turned off  1`] = `
{
  "definitions": {
    "child": {
      "properties": {
        "name": {
          "type": "string",
        },
        "parents": {
          "items": {
            "properties": {
              "children": {
                "items": [Circular],
                "type": "array",
              },
              "name": {
                "type": "string",
              },
            },
          },
          "type": "array",
        },
      },
    },
    "parent": {
      "properties": {
        "children": {
          "items": {
            "properties": {
              "name": {
                "type": "string",
              },
              "parents": {
                "items": [Circular],
                "type": "array",
              },
            },
          },
          "type": "array",
        },
        "name": {
          "type": "string",
        },
      },
    },
    "person": {
      "properties": {
        "name": {
          "type": "string",
        },
        "spouse": {
          "type": [Circular],
        },
      },
    },
    "thing": {
      "$ref": "#/definitions/thing",
    },
  },
}
`;
